daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -2364,6 +2358,35 @@ sealed abstract class DocumentPart(part: Node, parent: Document) {
     }
   }.trim.toUpperCase()
 
+  /*
+  * Allow "//" and "/* */" (inline) to act as comments.
+  * Any characters not explicitly allowed are also considered comments and are removed.
+  */
+  def removeComments(validCharactersSet : String, userData : String ) : String = {
+    lazy val doubleForwardPattern = "//.*".r
+    lazy val openClosePattern = "/[*].*[*]/".r
+
+    //Remove the comments (//) and (/* */)
+    lazy val noCommentsForward = doubleForwardPattern.replaceAllIn(userData,"")
+    lazy val noCommentsBothFormats = openClosePattern.replaceAllIn(noCommentsForward,"")
+
+    //Throw exception if /* or */ still found. This means user input was not formatted correctly.
+    if(noCommentsBothFormats.contains("/*") || noCommentsBothFormats.contains("*/")){
+      throw TDMLException("Improper formatting of /* */ style comment", None)
+    }
+
+    //Remove the new lines - concatenate after
+    lazy val noCommentsString = noCommentsBothFormats.split('\n').mkString
```

Comment:
Is it possible for `\r` to also exist in documentParts, like if running on windows? Or does it all get canonicalized to `\n`?

Reply:
So, in theory, XML loading should convert both \r\n and \r to lone \n. 

But, this has proven problematic for people like us, who want to use XML as a data language; hence, there is chatter on the internet (e.g., saw something about this on stack overflow) about options on loaders to shut this off.

I think the right thing is stick an Assert.invariant(...test..) check in the code that scans for \r characters and fails if any are detected. With a comment that says "// XML is not supposed to contain CR characters ever", that would make it clear without adding logic.